libs/test/SmartSyncTest/src/com/salesforce/androidsdk/smartsync/manager/SyncManagerTest.java

```diff
@@ -144,6 +149,20 @@ public void testSyncDownWithoutOverwrite() throws Exception {
         checkDb(idToFields, ACCOUNTS_SOUP);
     }
 
+    /**
+     * Test for sync down with metadata target.
+     */
+    @Test
+    public void testSyncDownForMetadataTarget() throws Exception {
+
+        // Builds metadata sync down target and performs sync.
+        trySyncDown(MergeMode.LEAVE_IF_CHANGED, new MetadataSyncDownTarget(Constants.ACCOUNT), ACCOUNTS_SOUP);
```

Comment:
In a real app, one would probably setup a soup to hold all the metadata.
Should you read it back and check a few keys on the json blob?

Reply:
Sure - I can enhance this test to add a few keys.

Reply:
Enhanced the test to verify the `keyPrefix` and `label`, which should be enough to ensure we're getting the correct object metadata.